internal/apiserver/backrestservice/backrestimpl.go

```diff
@@ -191,9 +191,19 @@ func CreateBackup(request *msgs.CreateBackrestBackupRequest, ns, pgouser string)
 
 		}
 
-		//get pod name from cluster
+		// get pod name from cluster
 		var podname string
-		podname, err = getPrimaryPodName(cluster, ns)
+		podname, err = getBackrestRepoPodName(cluster, ns)
+
+		if err != nil {
+			log.Error(err)
+			resp.Status.Code = msgs.Error
+			resp.Status.Msg = err.Error()
+			return resp
+		}
+
+		// check if primary is ready
+		err = isPrimaryReady(cluster, ns)
 
 		if err != nil {
```

Comment:
Do we still need this check? If so, I'd recommend:

```suggestion
		// check if primary is ready
		if err := isPrimaryReady(cluster, ns); err != nil {
```

If not, please just drop it.

Reply:
We do still want to do that check before taking the backup, but I will update it to the suggested change. Thanks!

Reply:
Yes, for taking the backup we need that check. For showing backups ,we don't need that check.